caillef/dustzh.lua

```diff
@@ -257,7 +263,49 @@ Client.Action3Release = function()
 end
 
 
+uiControls = {}
+local uiControlsMetatable = {
+    __index = {
+        _isInit = false,
+        init = function(self)
+            local bg
+            local ui = require("uikit")
+            bg = ui:createFrame(Color(0,0,0,0.5))
+            bg.Width = Screen.Width / 8
+            bg.Height = Screen.Height / 5
```

Comment:
Width & Height should be based on content, to perfectly encapsulate it:

<img width="260" alt="Screenshot 2023-07-21 at 08 08 00" src="https://github.com/cubzh/worlds/assets/6775074/a6ec39e0-4d32-4fe2-8be5-3bccbd8e20ab">

And it should be repositioned when screen is resized: 

<img width="1654" alt="Screenshot 2023-07-21 at 08 18 48" src="https://github.com/cubzh/worlds/assets/6775074/0bf61655-5da5-4585-a440-adefa227ba37">

A possible approach: 

```lua
local ui = require("uikit")
local bg = ui:createFrame(Color(0,0,0,0.5))

local welcomeText = ui:createText("Welcome to Dustzh!", Color.White)
welcomeText:setParent(bg)

local Action1Text = ui:createText("Action1: Jump", Color.White)
Action1Text:setParent(bg)

local Action2Text = ui:createText("Action2: Shoot", Color.White)
Action2Text:setParent(bg)

local Action3Text = ui:createText("Action3: Reload", Color.White)
Action3Text:setParent(bg)

local dismissText = ui:createText("(shoot to dismiss)", Color.White, "small")
dismissText:setParent(bg)

bg.parentDidResize = function()
    local padding = 4
    local maxWidth = math.max(welcomeText.Width, Action1Text.Width, Action2Text .Width, Action3Text.Width, dismissText.Width)
    local height = welcomeText.Height + padding + Action1Text.Height + padding + Action2Text.Height + padding + Action3Text.Height + padding + dismissText.Height

    bg.Width = maxWith + padding * 2
    bg.Height = height + padding * 2
    bg.pos = {Screen.Width * 0.5 - bg.Width * 0.5, Screen.Height * 0.5 - bg.Height * 0.5, 0}

    welcomeText.pos = {bg.Width * 0.5 - welcomeText.Width * 0.5, bg.Height - padding - welcomeText.Height, 0}
    Action1Text.pos = {bg.Width * 0.5 - Action1Text.Width * 0.5,  welcomeText.pos.Y - padding - Action1Text.Height, 0}
    Action2Text.pos = {bg.Width * 0.5 - Action2Text.Width * 0.5,  Action1Text.pos.Y - padding - Action2Text.Height, 0}
    Action3Text.pos = {bg.Width * 0.5 - Action3Text.Width * 0.5,  Action2Text.pos.Y - padding - Action3Text.Height, 0}
   dismissText.pos = {bg.Width * 0.5 - dismissText.Width * 0.5,  Action3Text.pos.Y - padding - dismissText.Height, 0}
end

```